lib/vmdb/loggers/log_device.rb

```diff
@@ -0,0 +1,36 @@
+module Vmdb::Loggers
+  class LogDevice < Logger::LogDevice
+    def initialize(log = nil, shift_age: nil, shift_size: nil, shift_period_suffix: nil, binmode: false, perm: 0o644, owner_uid: nil, owner_gid: nil)
+      @perm      = perm
+      @owner_uid = owner_uid
+      @owner_gid = owner_gid
+
+      super(log, :shift_age => shift_age, :shift_size => shift_size, :shift_period_suffix => shift_period_suffix, :binmode => binmode)
+    end
+
+    private
+
+    def create_logfile(filename)
+      begin
+        logdev = File.open(filename, (File::WRONLY | File::APPEND | File::CREAT | File::EXCL))
+        logdev.flock(File::LOCK_EX)
+        logdev.chmod(@perm) if @perm
+        logdev.sync = true
+        logdev.binmode if @binmode
+
+        begin
+          File.chown(@owner_uid, @owner_gid, filename) if @owner_uid || @owner_gid
+        rescue Errno::EPERM
+        end
+
+        add_log_header(logdev)
+        logdev.flock(File::LOCK_UN)
+      rescue Errno::EEXIST
+        # file is created by another process
+        logdev = open_logfile(filename)
+        logdev.sync = true
```

Comment:
I wanted to `super` this method but wasn't sure how this would play if the file existed already, maybe we do want to chmod/chown in that case also?